lib/gc.cpp

```diff
@@ -81,6 +81,33 @@ static void maybe_initialize_gc() {
     }
 }
 
+#ifdef __GLIBC__
+#define GNOEXCEPT __THROW
+#else
```

Comment:
It seems that this should depend on the version of glibc you're linking against as different version have different exception specifications?

Reply:
@ChrisDodd the `__THROW` specifier is defined as a macro in `cdefs.h` and appears to be whatever glibc wants to use in that case.

Reply:
@ChrisDodd and that `__THROW` macro has been in glibc since at least 1999, so it should work: https://github.com/bminor/glibc/blob/c1422e5b7cdb4400f934c91bcefa3a1a96d789fb/misc/sys/cdefs.h

Reply:
I would stick to `noexcept` to keep things simple? If the semantics of `__THROW` changed we can revisit.

Reply:
@fruffy I originally had it as `noexcept`, and I don't like `__THROW` because we're reaching using internal glibc macros, which I'm pretty sure we're not supposed to do.  So, I'm happy to change it back.